source/val/validation_state.cpp

```diff
@@ -831,6 +831,35 @@ bool ValidationState_t::IsPointerType(uint32_t id) const {
   return inst->opcode() == SpvOpTypePointer;
 }
 
+bool ValidationState_t::IncludesIntOrFloatOfWidth(uint32_t id, SpvOp check_type,
+                                                  uint32_t check_width) const {
+  if (!id) return false;
+  const Instruction* inst = FindDef(id);
+  assert(inst);
+  switch (inst->opcode()) {
+    case SpvOpTypeInt:
+    case SpvOpTypeFloat:
+    case SpvOpTypeBool:
+    case SpvOpTypeVector:
+    case SpvOpTypeMatrix:
+    case SpvOpTypeArray:
+      return GetComponentType(inst->id()) == check_type &&
```

Comment:
GetComponentType returns the result id of the component type, not the opcode.

Reply:
True! I think I fixed this and a couple other issues in the latest version